packages/frontend/src/hooks/contracts/useVaultManager.ts

```diff
@@ -1,66 +1,62 @@
+import { useAtomValue } from 'jotai'
+import { useUpdateAtom } from 'jotai/utils'
 
 import { toTokenAmount } from '@utils/calculations'
 import { OSQUEETH_DECIMALS } from '../../constants/'
 import { VAULTS_QUERY, VAULTS_SUBSCRIPTION } from '../../queries/squeeth/vaultsQuery'
 import { Vaults } from '../../queries/squeeth/__generated__/Vaults'
 import { squeethClient } from '../../utils/apollo-client'
-import { useAtomValue } from 'jotai'
 import { addressAtom, networkIdAtom } from 'src/state/wallet/atoms'
+import { vaultManagerPollingAtom } from 'src/state/positions/atoms'
 
 /**
  * get user vaults.
  * @param token token address
  * @param refetchIntervalSec refetch interval in seconds
  * @returns {Vault[]}
  */
-export const useVaultManager = () => {
-  const [vaults, setVaults] = useState<Array<any>>([])
+export const useVaultManager = (poll = false) => {
   const address = useAtomValue(addressAtom)
   const networkId = useAtomValue(networkIdAtom)
+  const setVaultManagerPolling = useUpdateAtom(vaultManagerPollingAtom)
 
-  const { data, loading, subscribeToMore } = useQuery<Vaults>(VAULTS_QUERY, {
+  const { data, loading, startPolling, stopPolling } = useQuery<Vaults>(VAULTS_QUERY, {
     client: squeethClient[networkId],
     fetchPolicy: 'cache-and-network',
     variables: {
       ownerId: address ?? '',
     },
   })
-  useEffect(() => {
-    subscribeToMore({
-      document: VAULTS_SUBSCRIPTION,
-      variables: {
-        ownerId: address ?? '',
-      },
-      updateQuery(prev, { subscriptionData }) {
-        if (!subscriptionData.data || subscriptionData.data.vaults.length === data?.vaults.length) return prev
-        const newVaults = subscriptionData.data.vaults
-        return { vaults: newVaults }
-      },
-    })
-  }, [address, subscribeToMore])
 
-  useEffect(() => {
-    ;(async () => {
-      if (!data?.vaults?.length) return
+  const currentVault = data?.vaults.find((vault) => new BigNumber(vault.collateralAmount).isGreaterThan(0))
+  const prevVault = usePrevious(currentVault)
 
-      const _vaults = data?.vaults
-        .filter((v) => {
-          return new BigNumber(v?.collateralAmount ?? 0).gt(0)
-        })
-        .map(({ id, NftCollateralId, collateralAmount, shortAmount, operator }) => ({
-          id,
-          NFTCollateralId: NftCollateralId,
-          collateralAmount: toTokenAmount(new BigNumber(collateralAmount), 18),
-          shortAmount: toTokenAmount(new BigNumber(shortAmount), OSQUEETH_DECIMALS),
-          operator,
-        }))
+  useEffect(() => {
+    if (poll && !prevVault && !currentVault) {
+      startPolling(500)
+    } else if (poll && new BigNumber(prevVault?.shortAmount).isEqualTo(new BigNumber(currentVault?.shortAmount))) {
+      startPolling(500)
+    } else {
+      stopPolling()
+      setVaultManagerPolling(false)
+    }
+  }, [currentVault?.shortAmount.toString(), poll, prevVault?.shortAmount.toString(), startPolling, stopPolling])
```

Comment:
Do we also put current collateral amount & previous collateral amount to dep array?

Reply:
Thinking we still need data?.vaults?.length in dep array and condition check, cuz if users open a 20 oSQTH vault with some collateral and close it, and then open a new vault with 30 oSQTH(open A vault and close A. and then open B vault, A != B), it will lead to the "else" in this if-else condition check, which seems to lead to stopPolling

Reply:
@yichiehliu so what happens usually is that if a vault is closed, the shortAmount will be 0. so if you open a "vault A" with 20oSQTH and close it, then open a "vault B" with 30osqth, A!= B and it will stop polling. I think it's okay for it to stop polling then since that will mean the vaultQuery has been updated. Please, let me know if I'm missing something